Assets/BossRoom/Scripts/Client/Game/State/ClientMainMenuState.cs

```diff
@@ -9,6 +19,93 @@ namespace Unity.Multiplayer.Samples.BossRoom.Client
+
+        private GameNetPortal m_GameNetPortal;
+
+        private ClientGameNetPortal m_ClientNetPortal;
+
+
+        [SerializeField] private GameObject[] _autoInjected;
+        private DIScope m_Scope;
+
+        [SerializeField] private NameGenerationData m_NameGenerationData;
+        [SerializeField] private LobbyUIMediator m_lobbyUIMediator;
+
+        [SerializeField] private CanvasGroup m_mainMenuButtons;
+        [SerializeField] private GameObject m_signInSpinner;
+
+        private void Awake()
+        {
+            m_mainMenuButtons.interactable = false;
+            m_lobbyUIMediator.Hide();
+            DIScope.RootScope.InjectIn(this);
+        }
+
+        [Inject]
+        private void InjectDependenciesAndInitialize(AuthenticationAPIInterface authAPIInterface)
+        {
+            m_Scope = new DIScope(DIScope.RootScope);
+
+            m_Scope.BindInstanceAsSingle(m_NameGenerationData);
+            m_Scope.BindInstanceAsSingle(m_lobbyUIMediator);
+
+            var unityAuthenticationInitOptions = new InitializationOptions();
+
+#if UNITY_EDITOR
+            if (ClonesManager.IsClone())
+            {
+                Debug.Log("This is a clone project.");
+                var customArguments = ClonesManager.GetArgument().Split(',');
+
+                //second argument is our custom ID, but if it's not set we would just use some default.
+
+                var hardcodedProfileID = customArguments.Length > 1 ? customArguments[1] : "defaultCloneID";
+
+                unityAuthenticationInitOptions.SetProfile(hardcodedProfileID);
+            }
+#endif
+            authAPIInterface.DoSignInAsync(OnAuthSignIn,  OnSignInFailed, unityAuthenticationInitOptions);
+
+            void OnAuthSignIn()
+            {
+                m_Scope.FinalizeScopeConstruction();
+
+                foreach (var go in _autoInjected)
+                {
+                    m_Scope.InjectIn(go);
+                }
+
+                m_mainMenuButtons.interactable = true;
+                m_signInSpinner.SetActive(false);
+
+                var localUser = m_Scope.Resolve<LobbyUser>();
+                var localLobby = m_Scope.Resolve<LocalLobby>();
+
+
+                Debug.Log($"Signed in. Unity Player ID {AuthenticationService.Instance.PlayerId}");
+
+                localUser.ID = AuthenticationService.Instance.PlayerId;
+                localUser.DisplayName = m_NameGenerationData.GenerateName();
+                // The local LobbyUser object will be hooked into UI before the LocalLobby is populated during lobby join, so the LocalLobby must know about it already when that happens.
+                localLobby.AddPlayer(localUser);
+            }
+
+            void OnSignInFailed()
+            {
+                Debug.LogError("For some reason we can't authenticate the user anonymously - that typically means that project is not properly set up with Unity services.");
```

Comment:
Say there's a 503 from the auth service, would we have this message as well? Could be misleading. Will there be other error messages coming from auth directly to help debug it?

Reply:
AuthenticationAPIInterface.ParseServiceException method will be called if there's an exception - it will log all the error information.